packages/wrangler/src/dev/miniflare/stdio.ts

```diff
@@ -61,9 +61,17 @@ function getProcessStreamDataListener(processStream: "stdout" | "stderr") {
 				logStructuredLog(structuredLog, processStream);
 			} else {
 				const level = processStream === "stdout" ? "log" : "error";
-				// Unexpectedly we've received a line that is not a structured log, so we simply
-				// log it to the most likely appropriate logger level
-				logger[level](line);
+				// Unexpectedly we've received a line that is not a structured log.
+				// Directly logging it would bypass the filtering in `logStructuredLog()`, so
+				// we construct a "fake" structured log
+				logStructuredLog(
+					{
+						timestamp: Date.now(),
+						level,
+						message: line,
+					},
+					processStream
+				);
```

Comment:
This is ultimately a bug in the runtime (fatal errors bypassing structured logging), but this workaround works better than our previous one.

cc @danlapid